### PR TITLE
12 create assessment put route

### LINKED ISSUE #12: Create a PUT route to update a users assessment with evidence
Our users will be wanting to submit evidence that they've completed an assessment.

Create a PUT route which allows them to do so. 

Something like /assessments/:id might work.

Refer to the [wiki](https://github.com/harakeke-2019/EDAMAME/wiki/Server-Routes) to see how the routes may need to be setup.

### PR DESCRIPTION
**Pull request by rebeccazhangnz**
Set up PUT route for /assessments/:id. Set up function which update evidence information from database.

closes #12

### CODE REVIEW

File: server/routes/starter.js [hunk 1]
```diff
@@ -6,4 +6,5 @@ router.get('/', (req, res) => {
   res.send('Welcome to the start of EDAMAME!')
 })
 
+
```
Comment (bryceknz) [inline server/routes/starter.js]: Why is this extra newline added?

File: server/routes/assessmentsRoute.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+const express = require('express')
+
+const assessmentsRouter = express.Router()
+
+router.get('/assessments/:id', (req, res) => {
+  const id = req.params.id
+})
+
+router.put('/assessments/:id', (req, res) => {
+  const assID = req.params.id
```
Comment (bryceknz) [inline server/routes/assessmentsRoute.js]: Can we change this variable name? 😂

We are in the assessments routes so I don't think prefixing it is necessary, and it would be inconsistent with the variable name on line 6.

Generally the shorthand is `id`, not `ID` as it not an acronym, and it can be difficult to read (e.g. https://stackoverflow.com/a/13862674).

File: server/routes/assessmentsRoute.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  const evidenceInfo = req.body
+  db.updateEvidence(assID, evidenceInfo).catch(err =>
+    res.status(500).json({ error: 'Fail to add evidence to assesment!' })
+  )
```
Comment (bryceknz) [inline server/routes/assessmentsRoute.js]: Small typo on line 13 - "assesment" should be "assessment".

File: server/assessmentsDb.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+const environment = process.env.NODE_ENV || 'development'
+const config = require('./knexfile')[environment]
+const connection = require('knex')(config)
+
+function getAssessments(id, db = connection) {}
+
+function updateEvidence(assId, evidenceInfo, db) {
```
Comment (bryceknz) [inline server/assessmentsDb.js]: I would prefer the following parameter names:
```
(id, evidence, db = connection)
```

**Review by bryceknz — CHANGES_REQUESTED** [review]
Please see comments for requested changes.

You will also need to update the Wiki with a description of the data structure returned by this new route :-)
https://github.com/harakeke-2019/EDAMAME/wiki/Server-Routes#-individual-assessment-view-

File: server/assessmentsDb.js (hunk 4 above)
Comment (rebeccazhangnz) [reply server/assessmentsDb.js]: updated variable names

**Comment by rebeccazhangnz** [comment]
Fixed merge conflict.  assessment PUT route and db function works, tested by postman. Updated wiki.

File: package.json [hunk 5]
```diff
@@ -7,10 +7,10 @@
   "author": "EDA",
   "license": "ISC",
   "scripts": {
-    "start": "webpack && node server",
-    "dev": "node server & webpack --watch",
+    "start": "concurrently \"webpack\" \"node server\"",
+    "dev": "concurrently \"nodemon server\" \"webpack --watch\"",
     "client": "webpack-dev-server --hot --inline",
-    "server": "node server",
+    "server": "nodemon server",
```
Comment (bryceknz) [inline package.json]: None of these changes appear to be related to this ticket. If you think they are required, please create a new ticket for this work :)

File: tests/db/assessmentsDB.test.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+const testEnv = require('./test-environment')
+const db = require('../../server/db/assessments')
```
Comment (bryceknz) [inline tests/db/assessmentsDB.test.js:2]: This file path doesn't exist.

**Review by bryceknz — CHANGES_REQUESTED** [review]
Some changes required before I can test if this piece is working.

File: tests/db/assessmentsDB.test.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+const testEnv = require('./test-environment')
+const db = require('../../server/db/assessmentsDb')
+
+beforeEach(() => {
+  testDb = testEnv.getTestDb()
```
Comment (bryceknz) [inline tests/db/assessmentsDB.test.js:7]: You haven't defined the variable `testDb` before this line, hence the error `testDb is not defined`.

**Review by bryceknz — DISMISSED** [review]

File: tests/db/assessmentsDB.test.js (hunk 7 above)
Comment (rebeccazhangnz) [reply tests/db/assessmentsDB.test.js:7]: added let testDb = null at before line 4.

File: package.json [hunk 8]
```diff
@@ -11,6 +11,7 @@
     "dev": "concurrently 'yarn client' 'yarn server'",
     "client": "webpack --watch",
     "server": "nodemon server --ignore server/public",
+
```
Comment (emilyparkes) [inline package.json]: Why was this line added?

**Review by emilyparkes — CHANGES_REQUESTED** [review]

File: package.json (hunk 8 above)
Comment (emilyparkes) [reply package.json]: Than I can accept :)

**Review by emilyparkes — CHANGES_REQUESTED** [review]

**Review by emilyparkes — APPROVED** [review]
